Simplified login: Update store picker with entry point to store creation

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7902 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds updates to the store picker for the simplified login project:
- Optional borders in the account header of the store picker
- New button "Add a Store" as the footer of the store list
- Action sheet for "Create a new store" and "Connect an existing store".
- Semi-bold font for the message on the empty store picker along with more spacing between the image and the text.
- "Add a Store" button at the bottom of the empty store picker.
- Tracks for taps on the new buttons. Also replaced event `site_picker_enter_store_address_tapped` with `site_picker_connect_existing_store_tapped`.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
With the `.simplifiedLoginFlowI1`  and `.storeCreationMVP` feature flags enabled:
1. Log out of the app if necessary.
2. On the prologue screen, select "Log In"
3. Log in with a valid WP.com email address.
4. Make sure that the store picker look matches the Figma design here: ERGtCFLyiC9CFpHsw1XIid-fi-392%3A3599.
5. Tap "Add a Store" button at the bottom of the screen if the store list is empty, or at the footer of the list. Notice in Xcode console: `🔵 Tracked site_picker_add_a_store_tapped`
6. On the action sheet that appears, tap "Create a new store", notice in Xcode console: `🔵 Tracked site_picker_create_store_tapped`.
7. Dismiss the store creation flow, tap the "Connect existing store" button from the action sheet, notice in Xcode console: `🔵 Tracked site_picker_connect_existing_store_tapped`.

With `.simplifiedLoginFlowI1`  enabled and `.storeCreationMVP` disabled:
- Repeat steps 1-5 in the last case. Notice that when the "Add a Store" button is tapped, you are navigated directly to the site discovery flow.

With `simplifiedLoginFlowI1` disabled: the store picker should look and work exactly as before. The only difference is that tapping on the "Enter Your Store Address" will track `site_picker_connect_existing_store_tapped`.

### Screenshots
<!-- Include before and after images or gifs when appropriate. -->
<img src="https://user-images.githubusercontent.com/5533851/198543778-b8e84a98-8020-4a95-81df-0a3e144562ad.png" width=320 />
<img src="https://user-images.githubusercontent.com/5533851/198543784-40cff2d6-5003-4f9a-bab0-982bc17179a0.png" width=320 />
<img src="https://user-images.githubusercontent.com/5533851/198543789-263445f9-4a85-4b21-856a-b71575d4b853.png" width=320 />


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

itsmeichigo committed:
Add optional borders for the store picker header view

itsmeichigo committed:
Update the account header of the store picker with the border for simplified login

itsmeichigo committed:
Add the add store button at the bottom of the store list

itsmeichigo committed:
Merge branch 'issue/7899-simplified-prologue' into issue/7902-store-picker-for-simplified-login

itsmeichigo committed:
Update title of the try another account button

itsmeichigo committed:
Remove create store button

itsmeichigo committed:
Add action sheet for the Add a store button

itsmeichigo committed:
Show the Add store button at the end of the store list only

itsmeichigo committed:
Update style for the empty store cell

itsmeichigo committed:
Update buttons on the empty store picker

itsmeichigo committed:
Update tap area for the Add store button

itsmeichigo committed:
Show site discovery when tapping the add store button with store creation is disabled

itsmeichigo committed:
Add tracks for new buttons in the store picker

hichamboushaba commented on WooCommerce/Classes/Authentication/Epilogue/StorePickerViewController.swift:
```diff
@@ -244,13 +261,15 @@ private extension StorePickerViewController {
             return
         }
 
+        accountHeaderView.updateContainerView(hasBorders: isSimplifiedLogin)
```
@itsmeichigo I see you are not adding the borders to the user info part when simplified login is not enabled.

I'm doing it for all cases, I did this because this the sense that I got from discussing this with Joe p1666708196780059-slack-C046HDZL87J, WDYT? Should we confirm this point too?

itsmeichigo replied:
Good point 👍 I asked Joe on the Figma design ERGtCFLyiC9CFpHsw1XIid-fi-392%3A3599.

itsmeichigo replied:
He replied that we should apply this to all variants so I'll update it on my PR.

itsmeichigo committed:
Replace site_picker_enter_store_address_tapped with site_picker_connect_existing_store_tapped

itsmeichigo committed:
Merge branch 'trunk' into issue/7902-store-picker-for-simplified-login

selanthiraiyan commented:
>With simplifiedLoginFlowI1 disabled: the store picker should look and work exactly as before. The only difference is that tapping on the "Enter Your Store Address" will track site_picker_connect_existing_store_tapped.

Should this "Enter Your Store Address" be displayed in the store picker screen when `simplifiedLoginFlowI1` is OFF and `storeCreationMVP` is ON? I couldn't find that button. Am I missing something here?

<img width="300" src="https://user-images.githubusercontent.com/524475/198932307-cbec0c51-192e-4d40-806b-339c7c79d4b1.png"/>

selanthiraiyan commented on WooCommerce/Classes/Authentication/Epilogue/EmptyStoresTableViewCell.swift:
```diff
@@ -12,8 +12,8 @@ final class EmptyStoresTableViewCell: UITableViewCell {
-            legendLabel.textColor = .textSubtle
-            legendLabel.font = .subheadline
+            legendLabel.textColor = .text
+            legendLabel.font = .font(forStyle: .title3, weight: .semibold)
```
Consider using `.title3SemiBold` from [here](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/Extensions/UIFont%2BHelpers.swift#L27-L29)

selanthiraiyan reviewed (approved):
Code and analytics LGTM. 🎉 

I left [a question](https://github.com/woocommerce/woocommerce-ios/pull/7965#issuecomment-1296533650) about a scenario in testing.

jaclync commented on WooCommerce/Classes/Analytics/WooAnalyticsStat.swift:
```diff
@@ -121,9 +121,11 @@ public enum WooAnalyticsStat: String {
     case sitePickerSiteDiscovery = "site_picker_site_discovery"
     case sitePickerNewToWooTapped = "site_picker_new_to_woo_tapped"
+    case sitePickerAddStoreTapped = "site_picker_add_a_store_tapped"
+    case sitePickerCreateNewStoreTapped = "site_picker_create_store_tapped"
```
this event is also part of the analytics plan in pe5sF9-I3-p2#comment-1147 - `site_picker_create_site_tapped` with a `source` property, which I implemented today in https://github.com/woocommerce/woocommerce-ios/pull/7985. Maybe you can leave this one out for now?

jaclync replied:
(It's already logged in Android)

itsmeichigo replied:
Sure, I'll remove my event. Thanks for letting me know!

itsmeichigo commented:
> Should this "Enter Your Store Address" be displayed in the store picker screen when simplifiedLoginFlowI1 is OFF and storeCreationMVP is ON? I couldn't find that button. Am I missing something here?

@selanthiraiyan That's a valid question! I'm asking Joe in a Slack thread here p1667189100766619-slack-C046HDZL87J - hopefully we can use the same UI for both variants. This PR will be on hold until we receive a confirmation on that.

itsmeichigo committed:
Remove `site_picker_create_store_tapped` and wait to use the event from another PR instead

itsmeichigo committed:
Use a helper font for the empty store label

itsmeichigo committed:
Make the border around the account header view available to all variants

itsmeichigo committed:
Remove check for simplified login feature flag in the store picker

itsmeichigo committed:
Remove New to Woo button from the store picker

itsmeichigo commented:
@selanthiraiyan Since Joe's confirmed in p1667204037373789/1666934058.542029-slack-C046HDZL87J that all changes made to the store picker should be applied to all variants, I've added more changes to remove the check for `.simplifiedLoginI1` feature flag in the store picker. 

I've also removed the tracking event `site_picker_create_store_tapped` since @jaclync is adding a similar one in #7985. This part will be updated later either on this PR or that one depending on which one gets merged first.

This PR is ready for another look!

selanthiraiyan reviewed (approved):
Thanks for the details, Huong!

This still LGTM. 

🚢
